stl/inc/charconv

```diff
@@ -1710,23 +1711,59 @@ _NODISCARD from_chars_result _Ordinary_floating_from_chars(const char* const _Fi
         return {_Next, errc{}};
     }
 
-    // Before we adjust the exponent, handle the case where we detected a wildly
-    // out of range exponent during parsing and clamped the value:
-    if (_Exponent > _Maximum_temporary_decimal_exponent) {
-        _Assemble_floating_point_infinity(_Fp_string._Myis_negative, _Value);
-        return {_Next, errc::result_out_of_range}; // Overflow example: "1e+9999"
+    // Handle exponent of an overly large absolute value.
+    if (_Exp_abs_too_large) {
+        if (_Exponent > 0) {
+            _Assemble_floating_point_infinity(_Fp_string._Myis_negative, _Value);
+            return {_Next, errc::result_out_of_range};
+        } else {
+            _Assemble_floating_point_zero(_Fp_string._Myis_negative, _Value);
+            return {_Next, errc::result_out_of_range};
+        }
     }
 
-    if (_Exponent < _Minimum_temporary_decimal_exponent) {
-        _Assemble_floating_point_zero(_Fp_string._Myis_negative, _Value);
-        return {_Next, errc::result_out_of_range}; // Underflow example: "1e-9999"
+    // Adjust _Exponent and _Exponent_adjustment when they have different signedness to avoid overflow.
+    if (_Exponent > 0 && _Exponent_adjustment < 0) {
+        if (_Is_hexadecimal) {
+            const ptrdiff_t _Further_adjustment = (_STD max)(-((_Exponent - 1) / 4 + 1), _Exponent_adjustment);

```

Comment:
Not worth resetting testing, but it'd be nice in these "`_Further_adjustment` is negative" cases, to instead negate the other term and use `(_STD min)`; this would imo, make it much clearer what's happening, and you won't be adding and subtracting a negative number.